modules/nw-ovn-k-anp-best-practices.adoc

```diff
@@ -0,0 +1,36 @@
+
+* There is only one `BaselineAdminNetworkPolicy` for the cluster so be careful in defining it and you might consider using it as a default guardrail policy.
+
+* Since `0-100` is the supported priority range, it is always better when designing your ANP to use a middle range like `30-70` because this leaves some placeholder
+for priorities before and after. Even in the middle range, you might want to leave gaps so that as your infrastructure requirements evolve over time, you are able to insert new ANPs when needed at the right priority level. If you pack your ANPs, then you might need to recreate all of them to accommodate any changes in the future.
+
+* `networks` peer in egress rule is meant to select external destinations. It is recommended to use `namespaces` and `pods` peers to select in-cluster peers. Do not use `networks` for podIP level matching.
+
+* When using `0.0.0.0/0` to create a strong `DENY` policy, it is recommended that you check to make sure that you have higher priority `Allow` rules for essential traffic like towards DNS and KAPI-server. This ensures that you do not lock up your entire cluster leaving it in a non-functional state.
+
+* `Pass` is a special action used to delegate policy decision to the `NetworkPolicy` layer. If you might consider using this action when considering the interoperability between ANP and your `NetworkPolicy`.
+
+* Each rule can have up to 100 peers. Every peer is added to the same `Address_Set` object. Selectors across multiple rules should not overlap. This ensures that that the same IPs do not appear in multiple `Address_Set`, which can cause performance and scale scale limitations due to an OVN-Kubernetes limitation.
+
+* Using `namedPorts` is not performance efficient since each name has to get evaluated against the `subject` of the policy and `peer` pod's container ports. You might use this with caution in larger clusters where you have many pods and each pod has many containers.
+
+[id="differences-with-netpol_{context}"]
+= Differences to consider between NetworkPolicy and ANP
+
+* API semantics for ANP are very different from `NetworkPolicy`. There is no implicit deny for ANP when you create the policy. You have to explicitly define what you need. As a consequence, you cannot use ANP to isolate a pod from everything for ingress traffic. You You might consider using `Deny` action to `0.0.0.0/0` and it will act like a default deny to everything or you could `Deny` from all other pods for ingress traffic.
+
+* Use the empty namespace selector with care for ANP because it will select _ALL_ namespaces in your cluster including the infrastructure namespaces like `kube-system`, `kapi-server`, and `openshift-*`. To secure your workloads, you might use workload specific labels and then use that as the namespace selector instead of using `{}` string. An empty namespace selector has the potential to lock up your entire cluster leaving it in a non-functional state.
```

Comment:
This is not even "with care". We _do not support_ applying ANPs to system namespaces, period. You are virtually gauranteed to eventually break you cluster if you do so.

Reply:
I tried to address this with a callout and rewording.